movie-app/src/containers/app/App.tsx

```diff
@@ -1,60 +1,124 @@
-  budget: 0,
-  revenue: 0,
-  runtime: 150,
-  genres: ["Action"]
-};
+import React, { useCallback, useEffect } from 'react';
+import { connect } from 'react-redux';
+import {
+  BrowserRouter as Router,
+  Redirect,
+  Route,
+  Switch,
+} from 'react-router-dom';
+import { IMovie } from '../../interfaces/IMovie';
+import {
+  getMovieById,
+  loadMovies,
+  setSearchParams,
+} from '../../redux/rootActions';
+import EmptyState from '../../shared-components/empty-state/EmptyState';
+import MovieItem from '../../shared-components/movie-item/MovieItem';
+import ErrorBoundary from '../error-boundary/ErrorBoundary';
+import HeaderContainer from '../header-container/HeaderContainer';
+import ListHeaderContainer from '../list-header-container/ListHeaderContainer';
+import MovieDetailsContainer from '../movie-details-container/MovieDetailsContainer';
+import MoviesListWrapper from '../movies-list-container/MoviesListWrapper';
+import SearchContainer from '../search-container/SearchContainer';
+import AppWrapper from './AppWrapper';
+import { useHistory, useLocation } from 'react-router-dom';
+import { useQuery } from '../../constants/constants';
 
-const App = () => {
-  const [moviesList, getMoviesList] = useState();
+const App = (props: any) => {
+  const history = useHistory();
+  const location = useLocation();
+  const query = useQuery(location.search);
 
-  const getMovies = () => {
-    const arr: IMovie[] = [];
-    for (let index = 0; index <= 10; index++) {
-      arr.push(MOVIE);
+  useEffect(() => {
+    if (!history.location.pathname.includes('film')) {
+      if (!props.movies.length) {
+        props.setSearchParams({
+          searchBy: query.get('searchBy'),
+          search: query.get('search'),
+          sortBy: query.get('sortBy'),
+        });
+      }
+      props.loadMovies();
     }
-    if (arr.length) {
-      return arr.map((item: IMovie) => {
-        return <MovieItem {...item} key={item.id + Math.random()}></MovieItem>;
-      });
-    } else {
-      return <EmptyState>No films found</EmptyState>;
+  }, [location]);
+
+  const navigate = (id: number) => {
+    history.push(`/film/${id}`);
+  };
+
+  const renderMovies = (movies: IMovie[], loading: boolean) => {
+    if (!loading) {
+      if (movies.length) {
+        return movies.map((item: IMovie) => {
+          return (
+            <MovieItem
+              navigate={() => navigate(item.id)}
+              {...item}
+              key={item.id + Math.random()}
+            ></MovieItem>
```

Comment:
don't use random as `keys`. `keys` should be predictable, so `id` is enough

Reply:
done